### PR TITLE
72-fix/pivot-table-ui-sum_aggregation-issue

### PR DESCRIPTION
**Pull request by Nitinyadav237**
I have solved #72 UI problem showing previous value and sum aggregation issue

**Description:**
              Resolved the UI issue displaying previous values by implementing unique keys that assist React in identifying changes, additions, or removals effectively. 
              Additionally, addressed the sum aggregation problem by explicitly converting 'value.column' to float.

https://github.com/pretzelai/pretzelai/assets/113238477/ca49e5d5-38d5-47d5-805f-786f806d6822

### CODE REVIEW

File: src/components/Pivot.tsx [hunk 1]
```diff
@@ -239,7 +241,7 @@ export default function PivotTable({ id }: { id: number }) {
                   {columns.map((column) => renderColumnBlock(column))}
                 </div>
               </div>
-              <div className="h-72 overflow-auto">
+              <div className="h-72 overflow-auto" key={JSON.stringify(gridData)}>
```
Comment (ramonverse) [inline src/components/Pivot.tsx:244]: can we use a better key than stringifying gridata? this is not performant at all

**Review by ramonverse — CHANGES_REQUESTED** [review]

File: src/components/Pivot.tsx (hunk 1 above)
Comment (Nitinyadav237) [reply src/components/Pivot.tsx:244]: image


Is it okay to use `uuidv4()` or should I use a different method?

Comment (ramonverse) [reply src/components/Pivot.tsx:244]: keys should be consistent between re-renders. Do a bit of research on this topic and update your PR :)

**Comment by ramonverse** [comment]
closing due to inactivity

**Pull request closed without merging**